Modules/Module_Processing/Module_Population/xASL_stat_GetROIstatistics.m

```diff
@@ -786,12 +793,38 @@
 					% x.S.DAT_Diff_CoV_PVC1(SubjSess,iROI) = xASL_stat_ComputeDifferCoV(DataIm, CurrentMask, 1, pvPrimary, [], 0);
 					% x.S.DAT_Diff_CoV_PVC2(SubjSess,iROI) = xASL_stat_ComputeDifferCoV(DataIm, CurrentMask, 2, pvPrimary, pvSecondary, 0);
 				% end
+				%% 4D temporal data calculations - do not always exist
+				if ~isempty(Data4D)
+					% Initialize the precalculated vectors
+					sCoV4D = zeros(1, size(Data4DIm, 2));
+					diffCoV4D = zeros(1, size(Data4DIm, 2));
+
+					% Precalculate the temporal values - use functions that work in 3D only to do it for each temporal point
+					for iRepetition=1:size(Data4DIm, 2)
+						sCoV4D(iRepetition) = xASL_stat_ComputeSpatialCoV(Data4DIm(:, iRepetition), CurrentMask, MinVoxels, 0);
+					end
 
+					if sum(CurrentMask(:)) > MinVoxels
+						CurrentMaskFull = xASL_im_Column2IM(CurrentMask, x.S.masks.WBmask);% We need the full mask for the Diff-CoV calculation

```

Comment:
Why don't we do this inside xASL_stat_ComputeDifferCoV ? To keep this code cleaner?

Reply:
Yes of course. MinVoxels is now inside.